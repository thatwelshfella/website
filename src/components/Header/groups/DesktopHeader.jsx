import React from 'react';
import PropTypes from 'prop-types';

import '../Header.less';

const DesktopHeader = ({ icon, menu }) => (
  <div className="header-component ">
    {icon}
    <div className="menu-items">
      {menu.map(items => (
        <a href={items.link} key={items.link}>
          {items.value.toUpperCase()}
        </a>
      ))}
    </div>
  </div>
);

DesktopHeader.propTypes = {
  icon: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
  menu: PropTypes.array.isRequired,
};

export default DesktopHeader;
