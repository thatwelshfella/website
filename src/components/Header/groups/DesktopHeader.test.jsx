import React from 'react';
import { shallow } from 'enzyme';
import DesktopHeader from './DesktopHeader';

describe('<DesktopHeader />', () => {
  const menu = [
    { link: '#home', value: 'Home' },
    { link: '#aboutMe', value: 'About me' },
    { link: '#timeline', value: 'Employment Timeline' },
    { link: '#github', value: 'Github Projects' },
  ];

  const cmp = shallow(<DesktopHeader icon="LD" menu={menu} />);

  describe('@renders', () => {
    it.only('should render with desktop header ', () => {
      expect(cmp).toMatchSnapshot();
    });
  });
});
