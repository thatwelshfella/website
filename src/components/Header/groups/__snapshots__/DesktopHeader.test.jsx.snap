// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DesktopHeader /> @renders should render with desktop header  1`] = `
<div
  className="header-component "
>
  LD
  <div
    className="menu-items"
  >
    <a
      href="#home"
      key="#home"
    >
      HOME
    </a>
    <a
      href="#aboutMe"
      key="#aboutMe"
    >
      ABOUT ME
    </a>
    <a
      href="#timeline"
      key="#timeline"
    >
      EMPLOYMENT TIMELINE
    </a>
    <a
      href="#github"
      key="#github"
    >
      GITHUB PROJECTS
    </a>
  </div>
</div>
`;
